coordinator/coordinator.go

```diff
@@ -168,42 +169,100 @@ func NewCoordinator(meta *metadata.Client, logger *zap.SugaredLogger, cli *clien
 		KVClient:   &kvc,
 		Spawner:    spawner,
 		Timeout:    600,
+		Storage: &ETCDStorage{
+			EtcdClient: cli,
+			KVClient:   &kvc,
+		},
 	}, nil
 }
 
 const MAX_ATTEMPTS = 2
 
-func (c *Coordinator) WatchForNewJobs() error {
-	c.Logger.Info("Watching for new jobs")
-	getResp, err := (*c.KVClient).Get(context.Background(), "JOB_", clientv3.WithPrefix())
+// type Executor interface {
+// 	ExecuteJob(store Storage)
+// }
+
+type Store interface {
+	GetJobs() ([]string, error)
+	CreateJob(key string, job metadata.CoordinatorJob) error
+	GetJob(key string) (metadata.CoordinatorJob, error)
+	DeleteJob(key string) error
+	LockJob(key string) error
+	UnlockJob(key string) error
+	HasLock(key string) (bool, error)
+}
+
+type ETCDStorage struct {
+	EtcdClient *clientv3.Client
+	KVClient   *clientv3.KV
+}
+
+func (e *ETCDStorage) GetJobs() ([]string, error) {
+	getResp, err := (*e.KVClient).Get(context.Background(), "JOB_", clientv3.WithPrefix())
 	if err != nil {
-		return fmt.Errorf("get existing etcd jobs: %w", err)
+		return nil, fmt.Errorf("get existing etcd jobs: %w", err)
```

Comment:
Is this that error that always says "Could not find key" and errors when it shouldnt?